### PR TITLE
Fix #2379: Make PreImageRevealInterval configurable

### LINKED ISSUE #2379: Make `PreImageRevealPeriod` configurable
It is currently hardcoded to 6, but should be a `validator` parameter with `6` as a default value.

### PR DESCRIPTION
**Pull request by Geod24**
Required a slight refactoring otherwise I would have had to change the interface of `getNextPreImage`, which proved very ugly and inconvenient.

### CODE REVIEW

**Comment by Geod24** [comment]
Added to the example config, originally missed it.

File: source/agora/consensus/protocol/Config.d [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import scpd.types.Stellar_SCP;
+import scpd.types.Stellar_types : NodeID;
+import scpd.types.Utils;
+
+import std.conv;
+
+/*******************************************************************************
+
+    Convert a QuorumConfig to the SCPQorum which the SCP protocol understands
```
Comment (linked0) [inline source/agora/consensus/protocol/Config.d]:
```suggestion
    Convert a QuorumConfig to the SCPQuorum which the SCP protocol understands
```

File: source/agora/consensus/protocol/Config.d [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        auto scp_quorum = toSCPQuorumSet(sub_quorum);
+        quorum.innerSets.push_back(scp_quorum);
+    }
+
+    return quorum;
+}
+
+/*******************************************************************************
+
+    Convert an SCPQorum to a QuorumConfig
```
Comment (linked0) [inline source/agora/consensus/protocol/Config.d]:
```suggestion
    Convert an SCPQuorum to a QuorumConfig
```

File: source/agora/consensus/protocol/Config.d [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    Params:
+        scp_quorum = the quorum config
```
Comment (linked0) [inline source/agora/consensus/protocol/Config.d]:
```suggestion
        scp_quorum = `SCPQuorumSet` instance
```

File: source/agora/consensus/protocol/Config.d [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    Returns:
+        `SCPQuorumSet` instance
```
Comment (linked0) [inline source/agora/consensus/protocol/Config.d]:
```suggestion
        a quorum config
```

**Review by linked0 — COMMENTED** [review]
Two typos and there were incorrect comments originally. It should be corrected in this PR.

**Comment by Geod24** [comment]
Thanks @linked0 , I went ahead and applied your suggestions and more.

**Review by linked0 — APPROVED** [review]
LGTM.
This makes testing code simpler.

**Pull request merged**